json-schema-cli/src/main/java/io/ballerina/jsonschema/cli/JsonSchemaCmd.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.jsonschema.cli;
+
+import io.ballerina.cli.BLauncherCmd;
+import io.ballerina.jsonschema.core.Generator;
+import io.ballerina.jsonschema.core.Response;
+import io.ballerina.jsonschema.core.SchemaUtils;
+import io.ballerina.projects.util.ProjectUtils;
+import picocli.CommandLine;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.Scanner;
+
+/**
+ * Main class to implement "json-schema" command for ballerina.
+ */
+@CommandLine.Command(name = "json-schema", description = "Generates Ballerina types for JSON-schema specification")
+public class JsonSchemaCmd implements BLauncherCmd {
+    private static final String CMD_NAME = "json-schema";
+
+    public static final String INVALID_BALLERINA_DIRECTORY_ERROR =
+            "Invalid Ballerina package directory: %s, cannot find 'Ballerina.toml' file.";
+    public static final String INVALID_DIRECTORY_PATH = "Error: Invalid directory path has been provided. "
+            + "Output path '%s' is a file";
+    public static final String OUTPUT_FILE_NAME = "types";
+    public static final String OUTPUT_FILE_EXTENSION = ".bal";
+    private static final String FILE_OVERWRITE_PROMPT = "The file '%s' already exists at %s. Overwrite? (Y/N): ";
+    private static final String AUTO_GENERATED_MESSAGE = "// AUTO-GENERATED FILE. DO NOT MODIFY.\n" +
+            "// This file is auto-generated by the Ballerina json-schema tool.";
+    private static final String EMPTY_STRING = "";
+
+    private final PrintStream outStream;
+    private final boolean exitWhenFinish;
+
+    public JsonSchemaCmd() {
+        this.outStream = System.err;
+        this.exitWhenFinish = true;
+    }
+
+    @CommandLine.Option(names = {"-h", "--help"}, usageHelp = true)
+    private boolean helpFlag;
+
+    @CommandLine.Parameters(description = "Input file path of the JSON schema", arity = "0..1")
+    private String inputPath = "";
+
+    @CommandLine.Option(names = {"-m", "--module"}, description = "The name of the module in which the Ballerina " +
```

Comment:
May be we can change this to `o`, `output`
@MaryamZi

Reply:
I believe this is for consistency with the XSD tool. And usually, module will make sense since it is not a standalone thing.